database/gorm/query.go

```diff
@@ -619,21 +638,47 @@ func (r *QueryImpl) With(query string, args ...any) ormcontract.Query {
 		case func(ormcontract.Query) ormcontract.Query:
 			newArgs := []any{
 				func(tx *gormio.DB) *gormio.DB {
-					query := arg(NewQueryWithWithoutEvents(tx, r.withoutEvents))
+					query := arg(NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config))
 
 					return query.(*QueryImpl).instance
 				},
 			}
 
 			tx := r.instance.Preload(query, newArgs...)
 
-			return NewQueryWithWithoutEvents(tx, r.withoutEvents)
+			return NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config)
 		}
 	}
 
 	tx := r.instance.Preload(query, args...)
 
-	return NewQueryWithWithoutEvents(tx, r.withoutEvents)
+	return NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config)
+}
+
+func (r *QueryImpl) refreshConnection(value any) {
+	model, ok := value.(ormcontract.ConnectionModel)
+	if !ok {
+		return
+	}
+
+	// Check if the model has a connection specified
+	if conn := model.Connection(); conn != "" && conn != r.instance.Name() {
```

Comment:
S: We can't use `conn` to equal `r.instance.Name()`, because `conn` is a connection name, not driver name, you need to get the driver name by `conn` through `r.config`. In addition, need to special judge `postgres` and `postgresql`, and to check the name of other DB drivers are consistent with us.